src/Common/tests/Collections/IListTest.cs

```diff
@@ -0,0 +1,1692 @@
+                {
+                    items = tempItems;
+                    Assert.Throws<ArgumentNullException>(
+                        () => list.Add(null));
+                }
+                else
+                {
+                    items = new object[33];
+                    list.Add(null);
+                    items[16] = null;
+                    Array.Copy(tempItems, 0, items, 0, 16);
+                    Array.Copy(tempItems, 16, items, 17, 16);
+                }
+                for (int i = 0; i < 16; i++)
+                {
+                    list.Add(tempItems[16 + i]);
+                }
+                CollectionAssert.Equal(items, list);
+            }
+        }
+
+        [Fact]
+        public void AddWithNullValueAtBeginning()
+        {
+            if (!ExpectedIsFixedSize && !ExpectedIsReadOnly)
+            {
+                var list = GetList(null);
+                object[] items;
+                if (ItemsMustBeNonNull)
+                {
+                    Assert.Throws<ArgumentNullException>(
+                        () => list.Add(null));
+                    items = GenerateItems(16);
+                    list.AddRange(items);
+                }
+                else
+                {
+                    items = new object[17];
+                    list.Add(null);
+                    items[0] = null;
+                    var tempItems = GenerateItems(16);
+                    list.AddRange(tempItems);
+                    Array.Copy(tempItems, 0, items, 1, 16);
+                }
+
+                CollectionAssert.Equal(items, list);
+            }
+        }
+
+        [Fact]
+        public void AddWithNullValueAtEnd()
+        {
+            if (!ExpectedIsFixedSize && !ExpectedIsReadOnly)
+            {
+                var list = GetList(null);
+                object[] items;
+                var tempItems = GenerateItems(16);
+                list.AddRange(tempItems);
+
+                if (ItemsMustBeNonNull)
+                {
+                    Assert.Throws<ArgumentNullException>(
+                        () => list.Add(null));
+                    items = tempItems;
+                }
+                else
+                {
+                    items = new object[17];
+                    Array.Copy(tempItems, 0, items, 0, 16);
+                    list.Add(null);
+                    items[16] = null;
+                }
+
+                CollectionAssert.Equal(items, list);
+            }
+        }
+
+        [Fact]
+        public void AddDuplicateValue()
+        {
+            if (!ExpectedIsFixedSize && !ExpectedIsReadOnly)
+            {
+                object[] items;
+                var list = GetList(null);
+                if (ItemsMustBeUnique)
+                {
+                    items = GenerateItems(16);
+                    list.AddRange(items);
+                    Assert.Throws<ArgumentException>(
+                        () => list.Add(items[0]));
```

Comment:
It was in the original tests and I like that it is explicit. Other tests may indirectly test this, but this one does so directly.